classes/module/Module.php

```diff
@@ -3497,6 +3505,38 @@ public function saveDashConfig(array $config)
     {
         return false;
     }
+
+    /**
+     * @param int $compatibility
+     *
+     * @return $this
+     */
+    public function setMultistoreCompatibility(int $compatibility): self
+    {
+        if (!in_array(
+            $compatibility,
+            [
+                self::MULTISTORE_COMPATIBILITY_NO,
+                self::MULTISTORE_COMPATIBILITY_NOT_CONCERNED,
+                self::MULTISTORE_COMPATIBILITY_PARTIAL,
+                self::MULTISTORE_COMPATIBILITY_YES,
+            ]
+        )) {
+            throw new PrestaShopException(sprintf('Value %s is not a valid multistore compatibility value', $compatibility));
+        }
+
+        $this->multistoreCompatibility = $compatibility;
+
+        return $this;
+    }
```

Comment:
IMO this shouldn't change on runtime
It should be set in the constructor, or as a public constant